backend/src/db/providers/authProvider.ts

```diff
@@ -0,0 +1,23 @@
+import UserModel from 'db/models/User';
+import { IUser } from 'interfaces/db/entities';
+
+const authProvider = async (login: string) => {
+  const dbUser = await UserModel.findOne({ username: login });
+  if (!dbUser) {
+    throw new Error('user not found');
+  }
+  return dbUser;
+};
+
+const saveTokenProvider = async (token: string, user: IUser) => {
+  await UserModel.updateOne(
+    { _id: user._id },
+    {
+      $set: {
+        refreshToken: token,
+      },
+    },
+  );
+};
```

Comment:
Well, this solution is pretty good. But you can simplify the proccess of updating user's refresh token using mongoose model's method `save()`. In case, when we already have user's mongoose model object, you can just:
```suggestion
        user.refreshToken = token; 
        await user.save();
```
It looks much more readable and it wont affect execution speed, but honestly I don't really know, how IUser interface will act in this case: will it allow use `.save()` or not. Anyway you can check this out 😄

Reply:
does in this case it ( .save() ) update the whole user document or only changed field ?

Reply:
cause in this example first conflits with reassign param in runtime
here this rule: https://eslint.org/docs/rules/no-param-reassign
so we should create copy of user like `const newUser = {...user}` and then do reassign

Reply:
Mongodb paradigm is about updating all the document even if only one field changed. But solution with reassigning is bad. If we can not cope with lint conflict in some other way, let's leave your code without changes